test/widgets/message_list_test.dart

```diff
@@ -52,7 +52,7 @@ void main() {
     UnreadMessagesSnapshot? unreadMsgs,
   }) async {
     addTearDown(testBinding.reset);
-    streams ??= subscriptions ??= [eg.subscription(eg.stream())];
+    streams ??= subscriptions ??= [eg.subscription(eg.stream(streamId: 123))];
```

Comment:
So it was kind of a latent bug already, or at least a wart, that these tests depended on this constant value. I think concretely they're depending on `eg.streamMessage` producing messages in the same stream that `eg.stream()` gives.

But I think if we make this a constant that this test file supplies, that actually makes that situation somewhat worse. It's now quite obscure where this value "123" is coming from or what it connects to — why it needs to be this instead of 234 or 1 or 1000. In the version in main, you can click through to `eg.stream` and see the 123 there, and at least what the test is depending on is only that you get that same value (from that same one line of code) repeatedly, rather than that you get 123 here and there's some other piece of code somewhere else (namely in `eg.streamMessage`) that also says 123 and the important thing is that this line matches that one.

A minimal fix for that problem would be for there to be some named value in `example_data.dart`, and then all the places where you're adding `streamId: 123` here can refer to that value by name. That recovers the property that it's clear what other pieces of code any given piece of code is counting on agreeing with about the stream ID.